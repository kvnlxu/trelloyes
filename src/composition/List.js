import React from 'react';
import './List.css';
import Card from './Card';

function List(props){
  return (
    <section className="List">
      <header className="List-header">
        <h2>{props.header}</h2>
      </header>
      {props.cards.map((card,i) => <Card title={card.title} content={card.content} key={i}></Card>)}
    </section>
  );
}

export default List;