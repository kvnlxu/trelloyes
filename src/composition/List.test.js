import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

describe('List', () => {

  it('renders without crashing', () => {
    // first create a DOM element to render the component into
    const div = document.createElement('div');

    //render the component, this is the actual test, if something is wrong it will fail here
    ReactDOM.render(<List header="foo" cards={[]}/>, div);

    //clean up code
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<List header="foo" cards={[]}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

});