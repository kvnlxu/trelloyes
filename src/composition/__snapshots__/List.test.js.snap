// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      foo
    </h2>
  </header>
</section>
`;

exports[`renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      foo
    </h2>
  </header>
</section>
`;
